### PR TITLE
Convert library to TS and add Web support

### PR DESCRIPTION
**Pull request by FiberJW**
## Why

I want this library to work in our universal apps on the web, and I want to be able to confidently make changes to this code with static type checking via TypeScript.

## How

This PR converts `index.js` and `upload-sourcemaps.js` to TypeScript and removes dead/inaccurate code.

This PR adds web support and wraps the re-exported Sentry Browser and Sentry React Native modules in their own variables, which is a breaking change.

This PR moves `upload-sourcemaps` to `dist/upload-sourcemaps`, so this is a breaking change that will involve the user changing their import to `sentry-expo/dist/upload-sourcemaps`

This also introduces a build step, but this step is automatically run with the `"prepare"` script.

## Test Plan

I'll be using this in my dogfood project, which will bring up any issues that I hope to fix before making this PR legit

### CODE REVIEW

**Comment by FiberJW** [comment]
I just rebased and successfully tested on a local project. I also added an entry to the 'unpublished' changelog section for you to check out/revise.

I also need to update sentry-expo with new upload-sourcemaps instructions https://docs.expo.io/guides/using-sentry/

File: src/index.ts [hunk 1]
```diff
@@ -117,11 +129,21 @@ class ExpoIntegration {
...
+  if (Platform.OS === 'web') {
+    return initBrowser({
+      ...(options as ExpoWebOptions),
+      enabled: __DEV__ ? options.enableInExpoDevelopment ?? false : true,
+    });
+  }
+
+  let optionsCopy = { ...options } as ExpoNativeOptions;
+
+  optionsCopy.integrations = [
```
Comment (cruzach) [inline src/index.ts:142]: can we rename this to `nativeOptions`? I added a commit that did this but I think your force-push overwrote it/subsequent commits 🙈

File: src/index.ts [hunk 2]
```diff
@@ -68,35 +87,28 @@ class ExpoIntegration {
...
       if (client && !__DEV__) {
-        client
-          .flush(client.getOptions().shutdownTimeout || 2000)
-          .then(() => {
-            defaultHandler(error, isFatal);
-          })
-          .catch((e) => {
-            logger.error(e);
-          });
+        client.flush(2000).then(() => {
```
Comment (cruzach) [inline src/index.ts:101]:
```suggestion
        // @ts-ignore PR to add this to types: https://github.com/getsentry/sentry-javascript/pull/2669
        client.flush(client.getOptions().shutdownTimeout || 2000).then(() => {
```

File: src/upload-sourcemaps.ts [hunk 3]
```diff
@@ -79,8 +97,6 @@ module.exports = async (options) => {
         '--ext',
         'map',
         '--rewrite',
-        '--strip-prefix',
```
Comment (cruzach) [inline src/upload-sourcemaps.ts:82]: think you may have rebased using an older commit? https://github.com/expo/sentry-expo/commit/21ed6d2a75633881881024ce72d66357a46a5da1

Comment (cruzach) [reply src/upload-sourcemaps.ts:82]: there are other instances like that, I'd re-pull master again and then rerebase

File: src/index.ts [hunk 4]
```diff
@@ -136,22 +158,22 @@ export const init = (options = {}) => {
     }),
   ];
 
-  if (!options.release) {
-    options.release = !!Constants.manifest
+  if (!optionsCopy.release) {
+    optionsCopy.release = !!Constants.manifest
       ? Constants.manifest.revisionId || 'UNVERSIONED'
       : Date.now();
```
Comment (cruzach) [inline src/index.ts:164]:
```suggestion
      : Date.now().toString();
```

File: src/index.ts [hunk 5]
```diff
@@ -27,36 +47,35 @@ class ExpoIntegration {
...
       if (Constants.manifest.publishedTime) {
-        Sentry.setTag('expoAppPublishedTime', Constants.manifest.publishedTime);
+        setTag('expoAppPublishedTime', Constants.manifest.publishedTime);
       }
     }
 
     if (Constants.sdkVersion) {
-      Sentry.setTag('expoSdkVersion', Constants.sdkVersion);
+      setTag('expoSdkVersion', Constants.sdkVersion);
     }
```
Comment (cruzach) [inline src/index.ts:76]: delete and replace with 
```
    if (!!Constants.manifest) {
      setTag('expoReleaseChannel', Constants.manifest.releaseChannel);
      setTag('appVersion', Constants.manifest.version ?? '');
      setTag('appPublishedTime', Constants.manifest.publishedTime);
      setTag('expoSdkVersion', Constants.manifest.sdkVersion);
    }
```

File: src/index.ts [hunk 6]
```diff
@@ -27,36 +47,35 @@ class ExpoIntegration {
...
       deviceYearClass: Constants.deviceYearClass,
       linkingUri: Constants.linkingUri,
     });
 
-    Sentry.setTags({
+    setTags({
       deviceId: Constants.installationId,
       appOwnership: Constants.appOwnership,
       expoVersion: Constants.expoVersion,
     });
```
Comment (cruzach) [inline src/index.ts:60]:
```suggestion
    setTags({
      deviceId: Constants.installationId,
      appOwnership: Constants.appOwnership,
    });

    if (Constants.appOwnership === 'expo' && Constants.expoVersion) {
      setTag('expoAppVersion', Constants.expoVersion);
    }
```

**Pull request merged**